src/solver.rs

```diff
@@ -94,36 +95,38 @@ pub fn resolve<P: Package, VS: VersionSet>(
             .should_cancel()
             .map_err(|err| PubGrubError::ErrorInShouldCancel(err))?;
 
-        log::info!("unit_propagation: {}", &next);
+        info!("unit_propagation: {}", &next);
         state.unit_propagation(next)?;
-        log::debug!(
+
+        debug!(
             "Partial solution after unit propagation: {}",
             state.partial_solution
         );
 
-        let potential_packages = state.partial_solution.potential_packages();
-        if potential_packages.is_none() {
-            drop(potential_packages);
-            // The borrow checker did not like using a match on potential_packages.
-            // This `if ... is_none ... drop` is a workaround.
-            // I believe this is a case where Polonius could help, when and if it lands in rustc.
-            return state.partial_solution.extract_solution().ok_or_else(|| {
-                PubGrubError::Failure(
-                    "How did we end up with no package to choose but no solution?".into(),
-                )
-            });
-        }
+        let potential_packages = match state.partial_solution.potential_packages() {
```

Comment:
Looking at the code as you have it here, it seems like a perfect place for the new `let ... else` syntax.

Reply:
Ha! Never used that one before. Pretty neat. I don't like how it makes the initial line quite long, but I do like how it removes one level of indentation from the body.

Reply:
Talking about syntax is so confusing. No there is a newer and even more concise syntax https://rust-lang.github.io/rfcs/3137-let-else.html

Something like
```rust
        let Some(packages) = state.partial_solution.potential_packages() else {
            return state.partial_solution.extract_solution().ok_or_else(|| {
                PubGrubError::Failure(
                    "How did we end up with no package to choose but no solution?".into(),
                )
            });
        };
```

Reply:
Sweet, I've just re-pushed it using that syntax :) That is definitely more concise.